libzeth/snarks/groth16/groth16_snark.hpp

```diff
@@ -28,45 +28,54 @@ template<typename ppT> class groth16_snark
     static keypair generate_setup(
         const libsnark::protoboard<libff::Fr<ppT>> &pb);
 
-    /// Generate the proof
+    /// Generate the proof (from the values set to the protoboard)
     static proof generate_proof(
         const libsnark::protoboard<libff::Fr<ppT>> &pb,
         const proving_key &proving_key);
 
+    /// Generate the proof (from given primary and auxiliary values)
+    static proof generate_proof(
+        const proving_key &proving_key,
+        const libsnark::r1cs_primary_input<libff::Fr<ppT>> &primary_input,
+        const libsnark::r1cs_auxiliary_input<libff::Fr<ppT>> auxiliary_input);
```

Comment:
The signature of this static member is slightly inconsistent with the one above (i.e. the `proving_key` is set as the first parameter, while it is the last one above). I'd recommend to keep "invariant parameters at the same place" for better consistency here, and thus modify
```cpp
    static proof generate_proof(
        const libsnark::protoboard<libff::Fr<ppT>> &pb,
        const proving_key &proving_key);
```
to
```cpp
    static proof generate_proof(
        const proving_key &proving_key
        const libsnark::protoboard<libff::Fr<ppT>> &pb);
```

Reply:
switched arg order